scripts/link-checker

```diff
@@ -51,7 +52,7 @@ check() {
 	link=$(echo "$1" | grep -oE "(https|http|ftp):[^|]*")
 	files=$(echo "$1" | grep -oE "\|.*" | sed 's/|/ /g')
 
-	if grep -Fxq "$link" "$WHITELIST"; then
+	if echo "$link" | grep -Eqxf "$WHITELIST"; then
```

Comment:
I think it makes sense to remove the option `x`:

>     -x, --line-regexp
>             Only input lines selected against an entire fixed string or regu-
>             lar expression are considered to be matching lines.

Dropping it should allow us to use simpler patterns like

```
github.com
```

in the whitelist instead of

```
https://github.com(\S)*
```

Reply:
While I would love a simpler syntax, wouldn't this also cause any other links like `https://example.com/fancy/github.com/stuff` to match?

Reply:
Of course, but I would assume we do not use such strange URLs in our documentation 😊. Anyway, the decision is up to you. If you do not want to allow partial matching, then you obviously do not have to implement it. It is also quite easy to add this feature afterwards, if we decide that the current system makes writing whitelist patterns to complicated

Reply:
Rethinking it, I changed it according to your suggestion but left a note in the whitelist docu. It's better if the syntax is simple. Thank you for your valuable review!